zmq/zmq.php

```diff
@@ -702,9 +702,9 @@ class ZMQDevice
      *
      * @param ZMQSocket $frontend Frontend parameter for the devices. Usually where there messages are coming.
      * @param ZMQSocket $backend Backend parameter for the devices. Usually where there messages going to.
-     * @param ZMQSocket $listener Listener socket, which receives a copy of all messages going both directions. The type of this socket should be SUB, PULL or DEALER.
+     * @param null|ZMQSocket $listener Listener socket, which receives a copy of all messages going both directions. The type of this socket should be SUB, PULL or DEALER.
      */
-    public function __construct(ZMQSocket $frontend, ZMQSocket $backend, ZMQSocket $listener = null)
+    public function __construct(ZMQSocket $frontend, ZMQSocket $backend, ?ZMQSocket $listener = null)
```

Comment:
As I wrote in one of the comments Nullable type hints can be added only in final classes and methods that can't be overridden. Otherwise, users who use PHP < 7.1 and override such methods will get warnings.  Class ZMQDevice isn't final so here should be no `?` in the type hint